core/domain/user_jobs_one_off_test.py

```diff
@@ -1585,6 +1585,272 @@ def test_one_setting_model_without_gae_id(self):
+            original_created_on_timestamp + datetime.timedelta(hours=1))
+        user_settings_model.last_created_an_exploration = (
+            original_created_on_timestamp + datetime.timedelta(hours=2))
+
+        final_created_on_timestamp = user_settings_model.last_updated
+        user_settings_model.put()
+
+        self.assertNotEqual(
+            user_settings_model.created_on, original_created_on_timestamp)
+        self.assertNotEqual(
+            user_settings_model.created_on, final_created_on_timestamp)
+
+        output = self._run_one_off_job()
+        # There already exists a UserSettings created in test_utils. It's
+        # corresponding user_auth_details_model is created first (which has
+        # a lower timestamp), hence it is also being updated here.
+        self.assertItemsEqual([['SUCCESS_UPDATED', 2]], output)
+
+        migrated_user_model = (
+            user_models.UserSettingsModel.get_by_id(self.USER_ID_1))
+        self.assertEqual(
+            migrated_user_model.created_on, final_created_on_timestamp)
+
+    def test_multiple_runs_of_one_off_job_raises_no_error(self):
+        user_settings_model = (
+            user_models.UserSettingsModel(
+                id=self.USER_ID_1,
+                email=self.EMAIL_1,
+            )
+        )
+        user_settings_model.update_timestamps()
+        original_created_on_timestamp = user_settings_model.created_on
+        user_settings_model.created_on += datetime.timedelta(hours=10)
+        final_created_on_timestamp = user_settings_model.last_updated
+        user_settings_model.put()
+
+        self.assertNotEqual(
+            user_settings_model.created_on, original_created_on_timestamp)
+        self.assertNotEqual(
+            user_settings_model.created_on, final_created_on_timestamp)
+
+        output = self._run_one_off_job()
+        # There already exists a UserSettings created in test_utils. It's
+        # corresponding user_auth_details_model is created first (which has
+        # a lower timestamp), hence it is also being updated here.
+        self.assertItemsEqual([['SUCCESS_UPDATED', 2]], output)
+        output = self._run_one_off_job()
+        self.assertItemsEqual([['SUCCESS_ALREADY_UP_TO_DATE', 2]], output)
+
+        migrated_user_model = (
+            user_models.UserSettingsModel.get_by_id(self.USER_ID_1))
+        self.assertEqual(
+            migrated_user_model.created_on, final_created_on_timestamp)
+
+    def test_update_created_on_works_correctly_for_multiple_users(self):
+        user_settings_model_1 = (
+            user_models.UserSettingsModel(
+                id=self.USER_ID_1,
+                email=self.EMAIL_1,
+            )
+        )
+        user_settings_model_1.update_timestamps()
+        original_created_on_timestamp_1 = user_settings_model_1.created_on
+        user_settings_model_1.created_on += datetime.timedelta(hours=10)
+        final_created_on_timestamp_1 = user_settings_model_1.last_updated
```

Comment:
Should we test other field of `UserSettingsModel` for the lowest value?

Reply:
You mean some other date-time attribute of `UserSettingsModel`? I printed out the value of all those attributes in this test, and found that only `created_on` and `last_updated` exist, and rest all are `None`. So that's why here only `last_updated` attribute could be the answer. Is this what you wanted me to verify?

Reply:
I mean to manualy set some other datatime field of `UserSettingsModel` to the lowest value to check that the extraction works from other fields works too.

Reply:
Okay, so I have created a separate test for that now - one test using `last_updated` and another using other attributes of `UserSettingsModel` for updating correct timestamp.